go/collectdtostatsd/main.go

```diff
@@ -0,0 +1,1744 @@
+package main
+
+import (
+	"context"
+	"crypto/tls"
+	"fmt"
+	"log"
+	"net"
+	"os"
+	"os/signal"
+	"regexp"
+	"strconv"
+	"strings"
+	"syscall"
+	"time"
+
+	"github.com/coreos/go-systemd/daemon"
+	"github.com/inverse-inc/packetfence/go/pfconfigdriver"
+
+	statsd "gopkg.in/alexcesaro/statsd.v2"
+	ldap "gopkg.in/ldap.v2"
+	"layeh.com/radius"
+	. "layeh.com/radius/rfc2865"
+)
+
+type TypeName struct {
+	Typename map[*regexp.Regexp]Types
+}
+
+type Types struct {
+	Type map[int]TypeDef
+}
+
+type TypeDef struct {
+	Name       string
+	DataSource StatsdType
+	Minimum    string
+	Maximum    string
+}
+
+type StatsdType interface {
+	Send(name string, a interface{})
+}
+
+var GAUGE gauge
+
+type gauge struct{}
+
+func (s gauge) Send(name string, a interface{}) {
+	c, err := statsd.New()
+	if err != nil {
+		log.Print(err)
+	}
+	defer c.Close()
+	c.Gauge(name, a)
+}
+
+var ABSOLUTE absolute
+
+type absolute struct{}
+
+func (s absolute) Send(name string, a interface{}) {
+	c, err := statsd.New()
+	if err != nil {
+		log.Print(err)
+	}
+	defer c.Close()
+	// c.Gauge(name, a)
+}
+
+var DERIVE derive
+
+type derive struct{}
+
+func (s derive) Send(name string, a interface{}) {
+	c, err := statsd.New()
+	if err != nil {
+		log.Print(err)
+	}
+	defer c.Close()
+	// c.Gauge(name, a)
+}
+
+var COUNTER counter
+
+type counter struct{}
+
+func (s counter) Send(name string, a interface{}) {
+	c, err := statsd.New()
```

Comment:
The statsd client gopkg.in/alexcesaro/statsd.v2 supports concurrency.

So it would be much more efficient to have this as a global variable and use the same client in all the goroutines